test/e2e/data/infrastructure-docker/main/cluster-template-dualstack/md-dualstack.yaml

```diff
@@ -0,0 +1,16 @@
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+kind: KubeadmConfigTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  template:
+    spec:
+      initConfiguration:
+        nodeRegistration:
+          kubeletExtraArgs:
+            node-ip: "0.0.0.0" # Note: The control plane node should use an ipv6 address as an internalIP. The worker node should use an ipv4 address as an internalIP.
```

Comment:
Kubernetes (kubeadm) doesn't allow setting this field as unspecified - i.e. `0.0.0.0` / `::` - for dualstack. The compromise here is to mix the nodes between ipv4 and ipv6.

Reply:
Be careful here https://github.com/kubernetes/kubernetes/pull/116305
There is problem with Dual stack node-ip with external cloud-provider, https://github.com/kubernetes/enhancements/issues/3705
It's fixed in k8s 1.27, added here https://github.com/kubernetes/enhancements/pull/3706/files/cf97fd57f3f0febdc24e211093707733d1ed6c64#diff-d14f8e17caa4cec27696e2ff5eeb00802582ecbf73e9b70684049a733700b44e
Before that, node will only show single IP stack IP with external cloud-provider. Please correct me if I'm wrong @christianang

Reply:
If I understand the issue correctly - setting this config on Kubelet means that the node object will only ever advertise one node address. Is that roughly right @lubronzhan?

I think, if we need to have explicitly dualstack nodes here, I can remove this flag and patch the node in the docker provider similar to what a cloud provider would do.

Reply:
Hi @killianmuldoon the issue is only with external cloud-provider. if it's baremetal it's fine. Will you set cloud-provider=external on kubelet when using docker provider? If not setting to external it's probably ok

Reply:
Yeah I think you both have it covered. As Lubron said it probably depends on if you set cloud-provider to external because in the end both affect the node-ips that are set in the node status.